### ISSUE TITLE
Reproducing RoBERTa Winogrande finetuning results

### ISSUE THREAD

**Issue by 8enmann** [pytorch/fairseq#1832]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
I'm having trouble reproducing the Winogrande results from the paper. Using the hyperparameters given in the README, I get ~77% val acc, which is good. However if I switch to `--wsc-cross-entropy` I get ~57% val acc. Is something about my setup broken? Or is finetuning very sensitive to hyperparameter settings?

#### Code

```sh
wget -nc https://dl.fbaipublicfiles.com/fairseq/models/roberta.large.tar.gz || true
tar -xzvf roberta.large.tar.gz
TOTAL_NUM_UPDATES=23750 # Total number of training steps.
WARMUP_UPDATES=2375     # Linearly increase LR over this many steps.
LR=1e-05                # Peak LR for polynomial LR scheduler.
MAX_SENTENCES=32        # Batch size per GPU.
SEED=1                  # Random seed.
ROBERTA_PATH=roberta.large/model.pt
PYTHONWARNINGS="ignore"  CUDA_VISIBLE_DEVICES=1 fairseq-train winogrande/ \
  --restore-file $ROBERTA_PATH \
  --reset-optimizer --reset-dataloader --reset-meters \
  --no-epoch-checkpoints --no-last-checkpoints --no-save-optimizer-state \
  --best-checkpoint-metric accuracy --maximize-best-checkpoint-metric \
  --valid-subset val \
  --fp16 --ddp-backend no_c10d \
  --user-dir $FAIRSEQ_USER_DIR \
  --task winogrande --criterion winogrande \
  --wsc-cross-entropy \
  --arch roberta_large --bpe gpt2 --max-positions 512 \
  --dropout 0.1 --attention-dropout 0.1 --weight-decay 0.01 \
  --optimizer adam --adam-betas '(0.9, 0.98)' --adam-eps 1e-06 \
  --lr-scheduler polynomial_decay --lr $LR \
  --warmup-updates $WARMUP_UPDATES --total-num-update $TOTAL_NUM_UPDATES \
  --max-sentences $MAX_SENTENCES \
  --max-update $TOTAL_NUM_UPDATES \
  --log-format simple --log-interval 100
```

#### What have you tried?
See above. I also had to rename the winogrande files since it looked for train and val instead of train_xl and dev.

#### What's your environment?
I had to use an earlier version of fairseq since master appears to be broken due to refactoring task/args spec.

 - fairseq Version (e.g., 1.0 or master): aa79bb9c37b27e3f84e7a4e182175d3b50a79041
 - PyTorch Version (e.g., 1.0): 1.4.0
 - OS (e.g., Linux): Linux xxx 5.0.0-1021-gcp #21~18.04.1-Ubuntu SMP Mon Oct 7 04:51:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): `pip install -e fairseq`
 - Python version: 3.7.3
 - CUDA/cuDNN version: Driver Version: 418.87.01    CUDA Version: 10.1
 - GPU models and configuration: V100 x 1
 - Any other relevant information: had to use an old version of apex since HEAD doesn't appear to support `fused_adam_cuda`

**Comment by myleott** [comment]
@ngoyal2707 can you take a look at this?

**Comment by ngoyal2707** [comment]
@8enmann as far as I remember, the difference in those 2 criterions is, for winogrande, the candidates are given and the task is binary classification task vs wsc the candidates are not give, so those are generated by named entities. 
The second task is much harder than the first one, I don't think, it's just matter of changing hyperparam, the task definition itself gets changed.

about the filenames mismatch, the authors of winogrande updated their paper and dataset after roberta was evaluated on the dataset.

**Comment by 8enmann** [comment]
I'm running on Winogrande in both cases, just changing the loss. The README says
```
  --task winogrande --criterion winogrande \
  --wsc-margin-alpha 5.0 --wsc-margin-beta 0.4 \
```
 I modified this to
```
  --task winogrande --criterion winogrande \
  --wsc-cross-entropy \
```
I get a huge difference in validation accuracy. My best guess is there is some hyperparameter such as learning rate that needs to be updated.

**Comment by ngoyal2707** [comment]
Sorry, I haven't tested winogrande in the setting you mentioned
